src/errors.ts

```diff
@@ -1,27 +1,23 @@
 import { t } from '@lingui/macro'
 
-export class IntegrationError extends Error {
-  constructor(message: string) {
-    super(message)
-    this.name = 'Integration Error'
-  }
-}
-
 export const DEFAULT_ERROR_HEADER = t`Please refresh the page and try again.`
 export const DEFAULT_ERROR_ACTION = t`Reload the page`
 
 export abstract class WidgetError extends Error {
-  header: string = DEFAULT_ERROR_HEADER
-  action: string = DEFAULT_ERROR_ACTION
+  header: string
+  action: string
 
   constructor(config: { header?: string; action?: string; message?: string }) {
     super(config.message)
-    if (config.header) {
-      this.header = config.header
-    }
-    if (config.action) {
-      this.action = config.action
-    }
+    this.header = config.header ?? DEFAULT_ERROR_HEADER
+    this.action = config.action ?? DEFAULT_ERROR_ACTION
+  }
+}
+
+export class IntegrationError extends WidgetError {
+  constructor(message: string) {
+    super({ message })
+    this.name = 'Integration Error'
```

Comment:
What does `this.name` do?

Reply:
this was my first time seeing this, and I just followed the `IntegrationError` example - but it seems like it's just [common practice ](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/name) and the names are probably shown in debugging tools sometimes.